nixos/modules/services/web-apps/lasuite-docs.nix

```diff
@@ -0,0 +1,473 @@
+            types.str
+            types.bool
+          ]
+        );
+        default = { };
+        example = ''
+          {
+            COLLABORATION_LOGGING = true;
+          }
+        '';
+        description = ''
+          Configuration options of collaboration server.
+
+          See https://github.com/suitenumerique/docs/blob/v3.1.0/docs/env.md
+        '';
+      };
+    };
+
+    domain = mkOption {
+      type = types.str;
+      description = ''
+        Domain name of the docs instance.
+      '';
+    };
+
+    config = mkOption {
+      type = types.attrsOf (
+        types.oneOf [
+          types.str
+          types.bool
+        ]
+      );
+      default = { };
+      example = ''
+        {
+          DJANGO_ALLOWED_HOSTS = "*";
+        }
+      '';
+      description = ''
+        Configuration options of docs.
+
+        See https://github.com/suitenumerique/docs/blob/v3.1.0/docs/env.md
+      '';
+    };
+
+    environmentFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        Path to environment file.
+
+        This can be useful to pass secrets to docs via tools like `agenix` or `sops`.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.lasuite-docs.config =
+      {
+        DJANGO_CONFIGURATION = mkDefault "Production";
+        DJANGO_SETTINGS_MODULE = mkDefault "impress.settings";
+        DATA_DIR = mkDefault "/var/lib/lasuite-docs";
+      }
+      // (optionalAttrs cfg.enableNginx {
+        DJANGO_ALLOWED_HOSTS = mkDefault "localhost,127.0.0.1,${cfg.domain}";
+      })
+      // (optionalAttrs cfg.database.createLocally {
+        DB_NAME = mkDefault "lasuite-docs";
+        DB_USER = mkDefault "lasuite-docs";
+        DB_HOST = mkDefault "/run/postgresql";
+      })
+      // (optionalAttrs cfg.redis.createLocally {
+        REDIS_URL = mkDefault "unix://${config.services.redis.servers.lasuite-docs.unixSocket}?db=1";
+        CELERY_BROKER_URL = mkDefault "redis+socket://${config.services.redis.servers.lasuite-docs.unixSocket}?db=2";
+      });
+
+    systemd.services.lasuite-docs = {
+      description = "Docs from SuiteNumérique";
+      after =
+        [ "network.target" ]
+        ++ (optional cfg.database.createLocally "postgresql.service")
+        ++ (optional cfg.redis.createLocally "redis-lasuite-docs.service");
+      wants =
+        (optional cfg.database.createLocally "postgresql.service")
+        ++ (optional cfg.redis.createLocally "redis-lasuite-docs.service");
+      wantedBy = [ "multi-user.target" ];
+
+      preStart = ''
+        ln -sf ${cfg.package}/lib/lasuite-docs/static /var/lib/lasuite-docs/
+
+        if [ ! -f .version ]; then
+          touch .version
+        fi
+
+        if [ "${cfg.package.version}" != "$(cat .version)" ]; then
+          ${getExe cfg.package} migrate && echo -n "${cfg.package.version}" > .version
+        fi
+      '';
+
+      script = ''
+        ${pythonPreloadSecrets}
```

Comment:
I think the secrets should go into `preStart` and the script can be replaced by `ExecStart` and use a structured approach to create the command line, given that users might want to customize the gunicorn commandline.

```
  ExecStart = utils.escapeSystemdExecArgs ([
    (lib.getExe' cfg.package "gunicorn")
    "--config=${gunicornSettingsFile"
    "impress.wsgi:application"
  ] ++ cfg.gunicorn.extraArgs);
```

Maybe appending `extraArgs` after the application module is too late, not sure.

Reply:
Wait until we have support for secret files and then use `serviceConfig.Environment`. We could even apply the patch temporarily.